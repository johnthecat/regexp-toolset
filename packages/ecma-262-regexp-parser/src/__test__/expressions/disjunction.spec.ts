import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

describe('Disjunction', () => {
  it('should parse simple disjunction', () => {
    expect(parseRegexp(/a|b/)).toMatchSnapshot();
  });

  it('should parse chained disjunction', () => {
    expect(parseRegexp(/a|b|c/)).toMatchSnapshot();
  });

  it('should parse disjunction with empty right size', () => {
    expect(parseRegexp(/a|/)).toMatchSnapshot();
  });

  // TODO fix position of zero length node
  it('should parse disjunction with empty left size', () => {
    expect(parseRegexp(/|b/)).toMatchSnapshot();
  });

  it('should parse disjunction inside group', () => {
    expect(parseRegexp(/(a|b)/)).toMatchSnapshot();
  });
});
