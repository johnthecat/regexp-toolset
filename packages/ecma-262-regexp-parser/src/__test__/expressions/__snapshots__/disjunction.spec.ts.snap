// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Disjunction > should parse chained disjunction 1`] = `
{
  "body": {
    "end": 5,
    "kind": 3,
    "left": {
      "end": 3,
      "kind": 3,
      "left": {
        "charCode": 97,
        "end": 1,
        "kind": 6,
        "start": 1,
        "type": "simple",
        "value": "a",
      },
      "right": {
        "charCode": 98,
        "end": 3,
        "kind": 6,
        "start": 3,
        "type": "simple",
        "value": "b",
      },
      "start": 1,
    },
    "right": {
      "charCode": 99,
      "end": 5,
      "kind": 6,
      "start": 5,
      "type": "simple",
      "value": "c",
    },
    "start": 1,
  },
  "end": 6,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Disjunction > should parse disjunction inside group 1`] = `
{
  "body": {
    "body": {
      "end": 4,
      "kind": 3,
      "left": {
        "charCode": 97,
        "end": 2,
        "kind": 6,
        "start": 2,
        "type": "simple",
        "value": "a",
      },
      "right": {
        "charCode": 98,
        "end": 4,
        "kind": 6,
        "start": 4,
        "type": "simple",
        "value": "b",
      },
      "start": 2,
    },
    "end": 5,
    "kind": 22,
    "specifier": null,
    "start": 1,
    "type": "capturing",
  },
  "end": 6,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Disjunction > should parse disjunction with empty left size 1`] = `
{
  "body": {
    "end": 2,
    "kind": 3,
    "left": {
      "end": 1,
      "expressions": [],
      "kind": 5,
      "start": 1,
    },
    "right": {
      "charCode": 98,
      "end": 2,
      "kind": 6,
      "start": 2,
      "type": "simple",
      "value": "b",
    },
    "start": 1,
  },
  "end": 3,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Disjunction > should parse disjunction with empty right size 1`] = `
{
  "body": {
    "end": 2,
    "kind": 3,
    "left": {
      "charCode": 97,
      "end": 1,
      "kind": 6,
      "start": 1,
      "type": "simple",
      "value": "a",
    },
    "right": {
      "end": 2,
      "expressions": [],
      "kind": 5,
      "start": 2,
    },
    "start": 1,
  },
  "end": 3,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Disjunction > should parse simple disjunction 1`] = `
{
  "body": {
    "end": 3,
    "kind": 3,
    "left": {
      "charCode": 97,
      "end": 1,
      "kind": 6,
      "start": 1,
      "type": "simple",
      "value": "a",
    },
    "right": {
      "charCode": 98,
      "end": 3,
      "kind": 6,
      "start": 3,
      "type": "simple",
      "value": "b",
    },
    "start": 1,
  },
  "end": 4,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;
